### PR TITLE
Minor updates to AI documentation

### PR DESCRIPTION
**Pull request by NathanJPhillips**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- N/A Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- N/A The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- N/A Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- N/A My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by owen-mc-diffblue** [comment]
Note that this probably has (fairly trivial) conflicts with the refactoring of AI in https://github.com/diffblue/cbmc/pull/5036

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by smowton — APPROVED** [review]

File: src/analyses/ai_domain.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ai_baset;
 class ai_domain_baset
 {
 protected:
-  /// The constructor is expected to produce 'false' or 'bottom'
+  /// The default constructor is expected to produce 'false' or 'bottom'
```
Comment (martin-cs) [inline src/analyses/ai_domain.h:30]: To be honest, it's all constructors.

**Review by martin-cs — APPROVED** [review]
These aren't wrong or unacceptable BUT there is a major refactoring patch to all of this code https://github.com/diffblue/cbmc/pull/5036 which will obsolete these comments as the template parameter will no longer be used like that, nor will we be limited to just default constructors.

How do you want me to proceed?

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0fab2ceaf3ed911023e35a2805263418a5733975).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124890633

**Comment by NathanJPhillips** [comment]
@martin-cs I'll hit the merge button now, understanding that this will be superseded by the new AI when it arrives.

**Pull request merged**